protocols/virtualdefund/virtualdefund.go

```diff
@@ -166,6 +169,14 @@ func (o *Objective) clone() Objective {
 	}
 	clone.MyRole = o.MyRole
 
+	// TODO: Properly clone the consensus channels
```

Comment:
I've started on #581. It might make sense to cherry pick [7b7724f](https://github.com/statechannels/go-nitro/pull/581/commits/7b7724f73ea1e49aec2ba977e8ec7b92ef8cde47) onto this branch, and then you can call the (unfinished at the moment) clone method here?

Reply:
Update: #581 is pretty much finished and is just pending a review.

Reply:
Awesome! I've approved #581, merged it, and rebased this PR.